models/errors.go

```diff
@@ -0,0 +1,79 @@
+package models
+
+import (
+	"context"
+	"errors"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+type ErrorList struct {
+	Errors []BaseError `json:"errors"`
+	Status int         `json:"-"`
+}
+
+type BaseError interface {
+	Error() string
+}
+
+type Error struct {
+	Cause       error  `json:"-"`
+	Code        string `json:"code"`
+	Description string `json:"description"`
+}
+
+func (e *Error) Error() string {
+	if e.Cause != nil {
+		return e.Cause.Error()
+	}
+	return e.Code + ": " + e.Description
+}
+
+func NewError(ctx context.Context, cause error, code string, description string) *Error {
+	log.Event(ctx, description, log.Error(cause), log.ERROR)
+	return &Error{
+		Cause:       cause,
+		Code:        code,
+		Description: description,
+	}
+}
+
+func NewValidationError(ctx context.Context, code string, description string) *Error {
+	log.Event(ctx, code+": "+description, log.ERROR)
```

Comment:
Minor point but our logging standards state we should avoid building dynamic strings for log events - we want the event string to be the same for every occurrence of a specific event. Can you update this to add the code the `log.Data` please? That way the event is the same regardless of the code and the code is additional data about the event.
````
log.Event(ctx, description, log.ERROR, log.Data{"code": code})
````